libs/wingsdk/src/target-sim/app.ts

```diff
@@ -265,6 +269,35 @@ export class App extends core.App {
     return this.outdir;
   }
 
+  /**
+   * Scans the app spec for token references and adds connections to reflect
+   * this relationship.
+   *
+   * @param spec The simulator spec
+   */
+  private addTokenConnections(spec: WingSimulatorSchema) {
+    const map: Record<string, IConstruct> = {};
+    for (const c of this.node.findAll()) {
+      map[c.node.path] = c;
+    }
+
+    for (const [from, resource] of Object.entries(spec.resources)) {
+      resolveTokens(resource.props, (to) => {
+        // skip references to the "handle" of the target resource because it would be reflected by
+        // the connections created by inflight method calls.
+        if (to.attr !== "handle") {
+          core.Connections.of(this).add({
+            source: map[from],
+            target: map[to.path],
+            targetOp: to.attr,
+            name: "<ref>",
+          });
```

Comment:
Food for thought -- this approach doesn't produce the "sourceOp" so we're losing information about what method(s) are referencing to the other resource. For example if we have a class like this:

```wing
bring cloud;

class A {
  bucket: cloud.Bucket;
  api: cloud.Api;
  new() {
    this.bucket = new cloud.Bucket();
    this.api = new cloud.Api();
  }
  pub inflight foo() {
    this.bucket.list();
  }
  pub inflight bar() {
    log(this.api.url);
  }
}

let a = new A();
```

Then we can show the specific relationship:

<img width="337" alt="Screenshot 2024-05-31 at 3 51 38 PM" src="https://github.com/winglang/wing/assets/5008987/e910a9f7-cd4e-434e-975b-4a404eca179f">

WDYT?

I think I got a minimal version working - here's a git patch if you feel up to incorporating it into your PR:

```patch
diff --git a/libs/wingsdk/src/std/resource.ts b/libs/wingsdk/src/std/resource.ts
index 3573d0d7e..7257da780 100644
--- a/libs/wingsdk/src/std/resource.ts
+++ b/libs/wingsdk/src/std/resource.ts
@@ -1,7 +1,9 @@
 import { Construct, IConstruct } from "constructs";
 import { App, LiftMap } from "../core";
 import { AbstractMemberError } from "../core/errors";
+import { parseToken } from "../simulator/tokens";
 import { Node } from "../std";
+import { isSimulatorToken } from "../target-sim/tokens";

 /**
  * A resource that can run inflight code.
@@ -219,6 +221,9 @@ function addConnectionsFromLiftMap(
   baseOp?: string
 ) {
   for (const [op, liftEntries] of Object.entries(liftData)) {
+    if (baseOp ?? op === "$inflight_init") {
+      continue;
+    }
     for (const [dep, depOps] of liftEntries) {
       if (Construct.isConstruct(dep) && !(dep instanceof AutoIdResource)) {
         // case 1: dep is an ordinary resource
@@ -234,6 +239,19 @@ function addConnectionsFromLiftMap(
       } else if (hasLiftMap(dep)) {
         // case 2: dep is an inflight
         addConnectionsFromLiftMap(construct, dep._liftMap, baseOp ?? op);
+      } else if (isSimulatorToken(dep)) {
+        // case 3: dep is a simulator token
+        const { path, attr, prop } = parseToken(dep);
+        const target = Node.of(construct)
+          .findAll()
+          .filter((c) => c.node.path === path)[0];
+        Node.of(construct).addConnection({
+          source: construct,
+          sourceOp: baseOp ?? op,
+          target,
+          targetOp: prop ?? attr,
+          name: "ref",
+        });
       }
     }
   }
```

Reply:
```sh
$ git apply <<HERE
PASTE THE ABOVE
HERE
```

Magic!

Reply:
@Chriscbr I am not sure how this would work... The lift map only includes information about inflight method calls, not references of preflight values. Right?

Reply:
I am leaving the previous implementation for now.

Reply:
> The lift map only includes information about inflight method calls, not references of preflight values. Right?

@eladb Here's the lift map for the example snippet above:

```
      get _liftMap() {
        return ({
          "foo": [
            [this.bucket, ["list"]],
          ],
          "bar": [
            [this.api.url, []],
          ],
          "$inflight_init": [
            [this.api.url, []],
            [this.bucket, []],
          ],
        });
      }
```
The token (this.api.url) is a reference to a preflight value and you can see it's included in the lift map -- that's how I was able to produce the screenshot in my earlier comment.

I think there's a bug in the UI where the console automatically displays everything with trailing parentheses, so it looks like `url()` instead of `url` in the screenshot - maybe that's the confusion?